sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/SclService.java

```diff
@@ -560,73 +559,65 @@ public static void removeAllControlBlocksAndDatasetsAndExtRefSrcBindings(final S
      * Activate used LDevice and Deactivate unused LDevice in {@link TLNode <em><b>TLNode </b></em>}
      *
      * @param scd SCL file for which LDevice should be activated or deactivated
-     * @return SclReport Object that contain SCL file and set of errors
+     * @return list of encountered errors
      */
-    public static SclReport updateLDeviceStatus(SCL scd) {
+    public static List<SclReportItem> updateLDeviceStatus(SCL scd) {
         SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
         SubstationAdapter substationAdapter = sclRootAdapter.getSubstationAdapter();
         final List<Pair<String, String>> iedNameLdInstList = substationAdapter.getIedAndLDeviceNamesForLN0FromLNode();
-        List<SclReportItem> errors = sclRootAdapter.streamIEDAdapters()
+        return sclRootAdapter.streamIEDAdapters()
                 .flatMap(IEDAdapter::streamLDeviceAdapters)
                 .map(LDeviceAdapter::getLN0Adapter)
                 .map(ln0Adapter -> ln0Adapter.updateLDeviceStatus(iedNameLdInstList))
                 .flatMap(Optional::stream)
-                .toList();
-        SclReport sclReport = new SclReport();
-        sclReport.getSclReportItems().addAll(errors);
-        sclReport.setSclRootAdapter(sclRootAdapter);
-        return sclReport;
+                .collect(Collectors.toList());
```

Comment:
Use `.toList()`. Caller should not assume that the returned List is modifiable.
Plus, the javadoc says : `Collectors.toList() : returns a Collector that accumulates the input elements into a new List. There are no guarantees on the type, mutability, serializability, or thread-safety of the List returned` so we should not assume that Collectors.toList() return a mutable List